source/extensions/filters/http/transformation/inja_transformer.h

```diff
@@ -115,7 +122,8 @@ class InjaTransformer : public Transformer {
 
   absl::optional<inja::Template> body_template_;
   bool merged_extractors_to_body_{};
-  Envoy::Random::RandomGenerator &rng_;
+  ThreadLocal::SlotPtr tls_;
+  std::unique_ptr<TransformerInstance> instance_;
```

Comment:
micro-nit: i don't think this needs to be a pointer
```suggestion
  TransformerInstance instance_;
```
(the .cc will change a bit as well)

Reply:
I followed the changes for this all the way until it needs a member function (`Environment::render`) within the Inja library to be marked `const`. I think this is because we're using it within the const `transform` method?

Reply:
It's unclear why the build errors occur when this field is a concrete object, but not when it is a pointer. However, the performance overhead should be negligent. We confirmed that the `render` calls are thread safe because the `Renderer` [constructor](https://github.com/pantor/inja/blob/master/single_include/inja/inja.hpp#L2679) that ends up being called by `Inja::Environment::render` takes `const` parameters. The same cannot be said of the `Parser` [constructor](https://github.com/pantor/inja/blob/master/single_include/inja/inja.hpp#L2031), but we should not be calling `parse` on the request path anyway.